fmp_server.py

```diff
@@ -305,18 +305,34 @@ def recvReqLayerUp(self, packet_id, data, seq):
          - Relocate > Select Server
         """
         if self.session.layer == 2:
-            city = self.session.get_city()
-            if city.leader == self.session:
-                if len(city.players) > 1:
-                    # TODO: Transfer the leadership to another player
-                    pass
-                city.leader = None
-
+            self.session.try_transfer_city_leadership()
             ntc_data = pati.lp2_string(self.session.capcom_id)
             self.server.layer_broadcast(self.session, PatID4.NtcLayerOut,
                                         ntc_data, seq)
         self.sendAnsLayerUp(data, seq)
 
+    def recvReqLayerEnd(self, packet_id, data, seq):
+        """ReqLayerEnd packet.
+
+        ID: 64020100
+        JP: レイヤ終了要求
+        TR: Layer end request
+        """
+        if self.session.layer > 1:
+            # City path
+            self.session.try_transfer_city_leadership()
+            ntc_data = pati.lp2_string(self.session.capcom_id)
+            self.server.layer_broadcast(self.session, PatID4.NtcLayerOut,
+                                        ntc_data, seq)
+            self.session.leave_city()
+        if self.session.layer > 0:
```

Comment:
I think this should be made into a `elif`, so the layer is not compared twice in case of being `> 1`

Reply:
This is actually the desired behavior (fallthrough)